Cookbook/Technical-Documents/BackendFeatureHowTo.md

```diff
@@ -0,0 +1,200 @@
+            method: .GET
+        )
+    }
+}
+```
+
+That's it!
+
+The `Service` usually provides `static` functions with parameters needed to create a request. In this case, we use `Void` in place of `Request` because it's just a `GET`, but we specify `CitiesResponse` as that's what we expect from the backend. 
+
+The URL is defined as a path, which is then constructed into a full URL with a helper enum. This accommodates the use of different environments while not requiring any changes from the clients.
+
+`ConcreteAuth` is the type of auth this endpoint expects as there are several, for different backends.
+
+Now that we have our `BackendResource`, we can define a *business controller* which will tie everything together and make the network call.
+
+However, first we need to learn about accessing backend resources.
+
+### Accessible
+`Accessible` is a protocol with just one function, which takes a backend resource and returns a signal producer with the parsed result:
+```
+func access<Request, Response>(_ resource: BackendResource<Request, Response, NoAuth>, with request: Request) -> SignalProducer<Response, CoreError>
+```
+
+`CoreError` is preferred as it allows both network and parsing errors and is used throughout the network layer.
+
+You may have noticed `NoAuth` in the signature – this is only good for unauthenticated requests, like fetching data before the user logs in. After that, you'll likely want to use `AuthenticatedAccessible` which provides a slightly different method:
+
+```
+func access<Request, Response>(_ resource: BackendResource<Request, Response, ConcreteAuth>, with request: Request) -> SignalProducer<Response, CoreError>
+```
+
+Using either `Accessible` or `AuthenticatedAccessible`, you can perform the network request defined by a backend resource, and this is exactly what we do in the business controller.
+
+### Business controller
+A `BusinessController` is the entity you will be interacting with in the view model, and as such, it needs to be injected in order to be stubbed for testing the view model, which means it is defined first as a protocol:
+
+```
+protocol CitiesBusinessControllerProtocol {
+    func fetch() -> SignalProducer<CitiesResponse, CoreError>
+}
+```
+
+Again, the business controller uses `fetch` terminology for getting some data from the backend. As this is the default and only method which gets no parameters, we call it simply `fetch()` without any modifiers.
+
+It returns a signal with a parsed response, which we can use in our view model directly.
+
+Let's define the concrete class for this protocol:
+```
+struct CitiesBusinessController: CitiesBusinessControllerProtocol {
+    private let accessible: AuthenticatedAccessible
+
+    init(accessible: AuthenticatedAccessible) {
+        self.accessible = accessible
+    }
+
+    func fetch() -> SignalProducer<CitiesResponse, CoreError> {
+        let resource = CitiesService.fetchCitiesResource()
+        return accessible.access(resource)
```

Comment:
I'm confused, isn't `AuthenticatedAccessible`'s `access` method expecting 2 parameters? I was not sure what the `with request: Request` was for when I saw your description of the protocol in the previous section, and since now in this example it's not even used, I'm not even sure if it's a typo or something I missed?

Reply:
This is a convenience for when there is no `Request` (which is *quite* often). I'll make a note about it.